lib/chef/knife/base_vsphere_command.rb

```diff
@@ -223,10 +223,22 @@ def find_folder(folderName)
         base_entity
       end
 
-      def find_network(networkName)
+      def find_network(networkName, dvswitch = nil)
         dc = datacenter
         base_entity = dc.network
-        base_entity.find { |f| f.name == networkName } || abort("no such network #{networkName}")
+
+        networks =
+          base_entity.select { |f| f.name == networkName } ||
+          abort("no such network #{networkName}")
+
```

Comment:
Hi, did you test this case where the network is not found? `#select` will return an empty array if nothing is found, which is truthy:

```Ruby
irb(main):006:0> [:a, :b, :c].select { |a| a == :d } || puts("not found")
=> []
```